src/witan/send/send.clj

```diff
@@ -135,7 +159,61 @@
                                                :count
                                                number-of-simulations))})
 
+(defn lag
+  "Dataset must be kept chronologically ordered in order to work correctly.
+   Shifts the specified column down by the specified number of rows, filling
+   in zeros at the top of the column and dropping the shifted values at the end."
+  ([dataset col-key]
+   (-> dataset
+       (ds/to-map)
+       (col-key)
+       (#(cons 0 %))
+       (butlast)))
+  ([dataset col-key lag-amount]
+   (-> dataset
+       (ds/to-map)
+       (col-key)
+       (#(concat (repeat lag-amount 0) %))
+       (as-> d (drop-last lag-amount d)))))
+
```

Comment:
Both the lamba and `as->` are here to work around the thread-first so I suggest a two-step approach instead